polygeist/test/polygeist-opt/sycl/matrix_multiply_reduction.mlir

```diff
@@ -26,7 +26,7 @@
 !sycl_accessor_write_1_ = !sycl.accessor<[1, f32, write, global_buffer], (!sycl_accessor_impl_device_1_, !llvm.struct<(memref<?xf32, 1>)>)>
 !sycl_accessor_read_1_ = !sycl.accessor<[1, f32, read, global_buffer], (!sycl_accessor_impl_device_1_, !llvm.struct<(memref<?xf32, 1>)>)>
 
-func.func private @matrix_multiply_reduction(%arg0: memref<?x!llvm.struct<(!sycl_accessor_write_1_, !sycl_accessor_read_1_, !sycl_accessor_read_1_)>, 4>, %arg1: i32) {
+func.func private @matrix_multiply_reduction(%arg0: memref<?x!sycl_accessor_write_1_, 4>, %arg1: memref<?x!sycl_accessor_read_1_, 4>, %arg2:memref<?x!sycl_accessor_read_1_, 4>, %arg3: i32) {
```

Comment:
After fixing the problem, some operations are unable to be hoisted. By splitting the memref struct argument like `ArgumentPromotion`, they can then be hoisted.